recipes/kubernetes_client_c/all/conanfile.py

```diff
@@ -0,0 +1,67 @@
+from conan import ConanFile
+from conan.tools.cmake import CMakeToolchain, CMake, cmake_layout, CMakeDeps
+from conan.tools.files import get, copy
+from os.path import join
+
+
+class kubernetes_client_cRecipe(ConanFile):
+    name = "kubernetes_client_c"
+    package_type = "library"
+
+    # Optional metadata
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/kubernetes-client/c"
+    description = "Official C client library for Kubernetes"
+    topics = ("kubernetes", "k8s", "kubernetes-client", "k8s-client")
+
+    # Binary configuration
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False], "curl_version": ["7", "8"]}
+    default_options = {"shared": False, "fPIC": True, "curl_version": "8"}
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            self.options.rm_safe("fPIC")
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.options["libwebsockets/*"].with_zlib = "zlib"
```

Comment:
We usually validate configurations if they are not valid instead of forcing the option like this directly. If you remove this line and compile with `-o="libwebsockets/*:with_zlib=False`, does it compile?

Reply:
Compiles without any issue